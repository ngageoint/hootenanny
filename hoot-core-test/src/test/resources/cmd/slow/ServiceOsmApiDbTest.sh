#!/bin/bash
set -e

#echo $PGDATABASE $PGHOST $PGPORT $PGUSER $PGPASSWORD

# clean out the database
source scripts/SetupOsmApiDB.sh

# setup DB variables for automation
source conf/DatabaseConfig.sh

export AUTH="-h $DB_HOST -p $DB_PORT -U $DB_USER"
export PGPASSWORD=$DB_PASSWORD

# setup dirs
rm -rf test-output/cmd/ServiceOsmApiDbTest
mkdir -p test-output/cmd/ServiceOsmApiDbTest

#echo $PGDATABASE $PGHOST $PGPORT $PGUSER $PGPASSWORD

# Load the database with known data
psql --quiet $AUTH -d $DB_NAME_OSMAPI -f hoot-core-test/src/test/resources/ToyTestA.sql

export DB_URL="postgresql://$DB_USER:$DB_PASSWORD@$DB_HOST:$DB_PORT/$DB_NAME_OSMAPI"

# do the read operation
echo "Performing read operation"
hoot convert -D services.db.writer.email=OsmApiInsert@hoot.local -D services.db.writer.create.user=true -D reader.add.source.datetime=false "$DB_URL" test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm

# create new maps file
scripts/generateIdMapXmlFromOsm test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm test-output/cmd/ServiceOsmApiDbTest/idmaps2.xml

# compare input to dumped
echo "Doing comparison"
scripts/compareOsmXmlToOsmApiDbDumpWriter test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm test-output/cmd/ServiceOsmApiDbTest/idmaps2.xml hoot-core-test/src/test/resources/ToyTestA.sql

