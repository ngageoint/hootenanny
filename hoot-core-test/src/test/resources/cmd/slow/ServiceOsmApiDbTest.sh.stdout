 setval 
--------
      2
(1 row)

 setval 
--------
      2
(1 row)

 setval 
--------
     37
(1 row)

 setval 
--------
      5
(1 row)

 setval 
--------
      1
(1 row)

Performing read operation
Doing comparison
OSM XML file:		test-output/cmd/ServiceOsmApiDbTest/ToyTestA-out.osm
PostgreSQL dump file:	hoot-core-test/src/test/resources/ToyTestA.sql

Validating data: nodes
	current_nodes, current_node_tags:
		Matching number of nodes: PASS (36)
		Deep inspection of data integrity: PASS
	nodes, node_tags:
		Matching number of nodes: PASS (36)
		Deep inspection of data integrity: PASS

Validating data: ways
	current_ways, current_way_tags, current_way_nodes:
		Matching number of ways: PASS (4)
		Deep inspection of data integrity: PASS
	ways, way_tags, way_nodes:
		Matching number of ways: PASS (4)
		Deep inspection of data integrity: PASS

Validating relations: ways
	current_relations, current_relation_tags, current_relation_nodes:
		Matching number of relations: PASS (0)
		Deep inspection of data integrity: PASS
	relations, relation_tags, relation_nodes:
		Matching number of relations: PASS (0)
		Deep inspection of data integrity: PASS
